lucene/core/src/java/org/apache/lucene/search/ConjunctionDISI.java

```diff
@@ -227,6 +236,7 @@ private int doNext(int doc) throws IOException {
 
   @Override
   public int advance(int target) throws IOException {
+    assertItersOnSameDoc();
```

Comment:
can we avoid iterating over all sub iterators unless assertions are enabled?

Reply:
addressed in 74151e3